### ISSUE TITLE
libbeat: can't override logging.files.name by command line option if filebeat.full.yml is loaded

### ISSUE THREAD

**Issue by cyrilleverrier** [elastic/beats#2918]
This bug is reproducible with any beats (filebeats, metrics beat, custom beat, etc.)

- Version: libbeat v5.0.0
- Operating System: Linux Ubuntu
- Steps to Reproduce: 

1. cd <...>/github.com/elastic/beats/filebeat
2. make
2. ./filebeat -c filebeat.yml -E logging.files.name=override1_filename.log
3. ./filebeat -c **filebeat.full.yml** -E logging.files.name=**override2_filename.log**
4. ls logs

Expected result:
In both cases, the log filename is overridden
```
$ls logs
 override1_filename.log override2_filename.log
```

Actual result:
The log filename is not overridden if the **filebeat.full.yml** is loaded
```
$ls logs
filebeat  override1_filename.log
```

There is a problem when merging the configuration from file and from command line.

**filebeat.full.yml** contains the section **logging.files** but is empty
**filebeat.yml** does not contain the section **logging.files**, and get default values

**Comment by ruflin** [comment]
@urso I think this is related to ucfg. Can you have a look?
@cyrilleverrier Thanks for reporting.

**Comment by cyrilleverrier** [comment]
note: @urso this bug is still present with ucfg v0.4.4

**Comment by urso** [comment]
Testing with console output only yet, using -E works for me.

I started PR #3304 adding debug output of config file loaded. This will help debugging the issue.

Will have a look at the described issue tomorrow.

**Comment by urso** [comment]
fix in progress: elastic/go-ucfg#85
